### PR TITLE
Add packaging info and scripts

### PR DESCRIPTION
**Pull request by StefansM**
This commit adds some documentation of the packaging process, some git hooks for safety and an autoupdate script for trivial version bumps.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,39 @@ See [@Pipeseroni/Packagers][packagers] for list of members who are responsible f
...
-TBA.
+After a new release is made of a Pipeseroni project, the following fields need
+to be updated:
+
+* `pkgver` in `PKGBUILD`.
+* `sha256sums` in `PKGBUILD` and `.SRCINFO`.
+
+### Automated packaging
+
+The entire packaging process can be automated with [scripts/autoupdate], which
```
Comment (livibetter) [inline README.md]: You should use [relative link](https://blog.github.com/2013-01-31-relative-links-in-markup-files/), I believe the following would work:

```markdown
... with [scripts/autoupdate](scripts/autoupdate), which
```

File: README.md [hunk 2]
```diff
@@ -23,7 +23,39 @@ See [@Pipeseroni/Packagers][packagers] for list of members who are responsible f
...
+for that project from GitHub's API. For branches with new releases, the script
+will bump `pkgver` and `sha256sums`, interactively commit, and prompt to push
+changes to the AUR.
+
+The autoupdate script requires that [`jq`][jq] be installed. It also requires
+`wget` and `bash`.
+
+### Git hooks
+
+It is recommended that you install [hooks/pre-commit] as a pre-commit hook (`cp
```
Comment (livibetter) [inline README.md]: Did you mean `scripts/...` here and in two more occurrences?

File: scripts/pre-commit [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <(printf "%s\n" "${pkgbuild_hashes[@]}") \
+    <(printf "%s\n" "${actual_hashes[@]}"))
+if [[ $? != 0 ]]; then
+    printf "PKGBUILD hashes do not match actual hashes. Diff:\n" >&2
+    printf "%s\n" "$diff_output" >&2
+fi
+
+# Check if arrays are equal. They must be the same length and contain equal
+# elements.
+[[ ${#pkgbuild_hashes[@]} == ${#actual_hashes[@]} ]] || hash_mismatch
```
Comment (livibetter) [inline scripts/pre-commit]: What is this `hash_mismatch`?  Or is this just intentionally to have a "command not found" ?

Also I think a simple:

```bash
[[ "${pkgbuild_hashes[@]}" == "${actual_hashes[@]}" ]] || hash_mismatch
```

should suffice the comparison without actually checking the length and comparing each hash individually.

**Comment by StefansM** [comment]
Wow, embarrassing. Yet more proof code reviews are a good thing. Thanks!

Also, at the top of `pre-commit` is an `exit 0`.

I will fix the scripts, test them more thoroughly, and update this PR.

File: README.md (hunk 1 above)
Comment (livibetter) [reply README.md]: You forgot this one.

**Review by livibetter — CHANGES_REQUESTED** [review]
Also update the Copyright section in README, and require all contribution to be under the same term, you might want to add another Contributing section.

**Review by livibetter — APPROVED** [review]

**Comment by livibetter** [comment]
@StefansM I'm merging this and the other two, because they were (almost) ready to be merged.  I hope I am not overstepping or interrupt what you still have any commits planned for them.

**Pull request merged**